src/NewsService.Business/FindNewsCommand.cs

```diff
@@ -16,38 +28,129 @@ public class FindNewsCommand : IFindNewsCommand
+        return null;
+      }
+
+      string errorMessage = "Cannot get authors now. Please try again later.";
+      const string logMessage = "Cannot get authors with ids: {authorIds}.";
+
+      try
+      {
+        object request = IGetUsersDataRequest.CreateObj(authorIds);
+        Response<IOperationResult<IGetUsersDataResponse>> response = _usersRequestClient
+          .GetResponse<IOperationResult<IGetUsersDataResponse>>(request)
+          .Result;
+
+        if (response.Message.IsSuccess)
+        {
+          return response.Message.Body.UsersData;
+        }
+
+        _logger.LogWarning(logMessage, string.Join(", ", authorIds));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage, string.Join(", ", authorIds));
+      }
+
+      errors.Add(errorMessage);
+
+      return null;
+    }
+
+    private List<DepartmentData> GetDepartments(List<Guid> departmentIds ,List<string> errors)
+    {
+      if (departmentIds == null || departmentIds.Count == 0)
+      {
+        return null;
+      }
+
+      const string errorMessage = "Can not get departments. Please try again later.";
+
+      try
+      {
+        object request = IGetDepartmentsRequest.CreateObj(departmentIds);
+        Response<IOperationResult<IGetDepartmentsResponse>> response = _departmentsRequestClient
+          .GetResponse<IOperationResult<IGetDepartmentsResponse>>(request)
+          .Result;
+
+        if (response.Message.IsSuccess)
+        {
+          return response.Message.Body.Departments;
+        }
+        else
+        {
+          _logger.LogWarning("Errors while getting departments. Reason: {Errors}",
+            string.Join('\n', response.Message.Errors));
+        }
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, errorMessage);
+      }
+
+      errors.Add(errorMessage);
+
+      return null;
+    }
 
     public FindNewsCommand(
       INewsRepository repository,
       INewsInfoMapper mapper,
-      IHttpContextAccessor httpContextAccessor)
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetDepartmentsRequest> departmentsRequestClient,
+      IRequestClient<IGetUsersDataRequest> usersRequestClient,
+      ILogger<GetNewsCommand> logger)
     {
       _repository = repository;
       _mapper = mapper;
       _httpContextAccessor = httpContextAccessor;
+      _departmentsRequestClient = departmentsRequestClient;
+      _usersRequestClient = usersRequestClient;
+      _logger = logger;
     }
 
     public FindResultResponse<NewsInfo> Execute(FindNewsFilter findNewsFilter)
     {
       FindResultResponse<NewsInfo> response = new();
 
-      response.Body = _repository
-        .Find(findNewsFilter, out int totalCount)
-        .Select(_mapper.Map)
-        .ToList();
+      List<DbNews> dbNewsList = _repository.Find(findNewsFilter, out int totalCount);
+
+      List<Guid> departmentsIds = dbNewsList.Select(d => d.DepartmentId.Value).ToHashSet().ToList();
```

Comment:
why do you convert it to hash set? you can use distinct

Reply:
done